src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,49 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.Workable;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    private MachineProducer bulldozerProducer = new BulldozerProducer();
+    private MachineProducer excavatorProducer = new ExcavatorProducer();
+    private MachineProducer truckProducer = new TruckProducer();
```

Comment:
The `MachineProducer` instances should be parameterized to ensure type safety. For example, use `MachineProducer<Truck>` for `truckProducer`.